### ISSUE TITLE
<mapDate> configuration option is ignored for nullable dates in Kotlin

### ISSUE THREAD

**Issue by jgerlier** [vojtechhabarta/typescript-generator#453]
I recently updated from version 2.18.565 to version 2.19.577 to benefits from the compatibility with the Kotlin nullability, and I noticed that for the nullable Kotlin fields with the `java.util.Date` type the `<mapDate>` configuration option is ignored.

For example :
```typescript
emailSentDate: DateAsString;
```
became
```typescript
emailSentDate?: Nullable<Date>;
```
instead of the expected
```typescript
emailSentDate?: Nullable<DateAsString>;
```

For now an simple workaround is to add :
```xml
<customTypeMappings>
    <customTypeMapping>
        java.util.Date:DateAsString
    </customTypeMapping>
</customTypeMappings>
```

By the way, thanks for your work on this library, it works really well !

**Comment by vojtechhabarta** [comment]
Thanks for this bug report with a good description.
I [commited](https://github.com/vojtechhabarta/typescript-generator/commit/775ca4c2c42b7efd340c9ff12b832448af7d7c11) fix.

**Comment by jgerlier** [comment]
Thanks a lot for your quick fix and response !

**Comment by vojtechhabarta** [comment]
Fix released in [2.20.583](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.20.583).
